api/src/main/java/io/grpc/ClientTransportHook.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2023 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+/**
+ * Listens on the client transport life-cycle events. These filters do not have the capability
+ * to modify the channels or transport life-cycle event behavior, but they can be useful hooks
+ * for transport observability. Multiple filters may be registered to the client.
+ */
+@ExperimentalApi("https://gitub.com/grpc/grpc-java/issues/TODO")
```

Comment:
Same as the other comment:

> This should be a new ticket with "experimental API"  label. Example: https://github.com/grpc/grpc-java/issues/10108.

I think it'd even make sense to use the same ticket to track both `ClientTransportHook` interface, and channel builder's `addTransportHook()`. 

@ejona86 - correct me if I'm wrong here